Import upstream commit:

commit 5070a06a9a7256ea66a49b534f734abc79ac100e
Author: Benjamin Dauvergne <bdauvergne@entrouvert.com>
Date:   Tue Sep 4 10:42:13 2018 +0200

    xml: fix parsing of saml:AuthnContext (fixes #25640)

    Decl/DeclRef are alternatives, when matching a Decl we should jump over
    the DeclRef.

--- a/lasso/xml/saml-2.0/saml2_authn_context.c
+++ b/lasso/xml/saml-2.0/saml2_authn_context.c
@@ -61,7 +61,7 @@
 static struct XmlSnippet schema_snippets[] = {
 	{ "AuthnContextClassRef", SNIPPET_CONTENT | SNIPPET_OPTIONAL | SNIPPET_JUMP_ON_MISS | SNIPPET_JUMP_3,
 		G_STRUCT_OFFSET(LassoSaml2AuthnContext, AuthnContextClassRef), NULL, NULL, NULL},
-	{ "AuthnContextDecl", SNIPPET_NODE | SNIPPET_OPTIONAL | SNIPPET_JUMP_ON_MISS | SNIPPET_JUMP_4,
+	{ "AuthnContextDecl", SNIPPET_NODE | SNIPPET_OPTIONAL | SNIPPET_JUMP_ON_MATCH | SNIPPET_JUMP_4,
 		G_STRUCT_OFFSET(LassoSaml2AuthnContext, AuthnContextDecl), NULL, NULL, NULL},
 	{ "AuthnContextDeclRef", SNIPPET_CONTENT | SNIPPET_OPTIONAL | SNIPPET_JUMP | SNIPPET_JUMP_3,
 		G_STRUCT_OFFSET(LassoSaml2AuthnContext, AuthnContextDeclRef), NULL, NULL, NULL},
